lldb/source/DataFormatters/FormatManager.cpp

```diff
@@ -222,14 +223,15 @@ void FormatManager::GetPossibleMatches(
 
   if (compiler_type.IsPointerType()) {
     CompilerType non_ptr_type = compiler_type.GetPointeeType();
-    GetPossibleMatches(valobj, non_ptr_type, use_dynamic, entries,
-                       current_flags.WithStrippedPointer());
+    if (dereference_ptr)
+      GetPossibleMatches(valobj, non_ptr_type, use_dynamic, entries,
+                        current_flags.WithStrippedPointer(), false);
```

Comment:
Are you sure we need the extra flag? Could we use `current_flags.stripped_pointer` as an indicator of whether we've removed a level of pointers?

Reply:
It's unnecessary. Removed.